h2o-core/src/main/java/hex/Model.java

```diff
@@ -747,6 +747,14 @@ public final String getTreatmentColumn(){
     public final int getMaxCategoricalLevels() {
       return _max_categorical_levels;
     }
+
+    public void setDistributionFamily(DistributionFamily distributionFamily){
+      _distribution = distributionFamily;
+    }
+
+    public DistributionFamily getDistributionFamily() {
+      return _distribution;
+    }
```

Comment:
shouldn't we make the `_distribution`, `_family`, `_link`... properties private now that we have those accessors? How otherwise can we enforce that those properties are always set through this setter?

Reply:
The `DistributionFamily` has no information of `link` as it is used only together with `family`. (The same goes for the distribution parameters (`twedie_power`, etc).) This is ok if we use it to set `family` inferred from `distribution` but it would be limiting for other use-cases. 

I think the benefit of this setter and getter is limited to AutoML and "compound" algos (SE, rulefit) so I wouldn't enforce its usage.